src/mailroom_madness.py

```diff
@@ -0,0 +1,267 @@
+
+    If name is list, list donor_data.
+    If name is not in donor_data, enter it with [] value.
+    Returns name.
+
+    Args:
+        donor_data (dict)
+    """
+    name = get_name(donor_data)
+    if not name:
+        return False
+    donor_data.setdefault(name, [])
+    return name
+
+
+def get_name(donor_data, out=sys.stdout):
+    """Get a name from the user.
+
+    Return the amount or false if the
+    user wishes to return to the main prompt.
+
+    Args:
+        donor_data (dict)
+        out (file, optional)
+    """
+    prompt = u'Enter a donor name, "list", or "cancel" > '
+    while True:
+        name = input(prompt)
+        if name == u'list':
+            list_donor_names(donor_data.keys(), out)
+        elif name == u'cancel':
+            return False
+        elif name:
+            return name
+        else:
+            out.write('Please give a name.')
+
+
+def list_donor_names(names, out=sys.stdout):
+    """Print a list of donor names.
+
+    Args:
+        names (list): List of donor names.
+        out (file, optional)
+    """
+    out.write(u'Donors:\n')
+    for name in sorted(names):
+        out.write(u'* {}\n'.format(name))
+
+
+def get_donation_amount(out=sys.stdout):
+    """Get an donation amount from the user.
+
+    Return the amount or
+    false if the user wishes to return to the main prompt.
+
+    Args:
+        out (file, optional): Description
+    """
+    prompt = u'Enter a donation amount or "cancel"> '
+    failure = u"That's not a positive integer."
+    while True:
+        user_input = input(prompt)
+        if user_input == u'cancel':
+            return False
+        elif validate_donation_amount(user_input):
+            return int(user_input)
+        else:
+            out.write(u'{}\n'.format(failure))
+
+
+def validate_donation_amount(user_input):
+    """Take user input, returns boolean."""
+    try:
+        return int(user_input) > 0
+    except ValueError:
+        return False
+
+
+def send_thank_you(donor_data):
+    """Prompt user for a name and donation amount and records them.
+
+    Args:
+        donor_data (dict)
+    """
+    name = prompt_donor_name(donor_data)
+    if name is False:
+        return
+
+    amount = get_donation_amount()
+    if amount is False:
+        return
+
+    log_thank_you(name, amount, donor_data)
+
+
+def log_thank_you(name, num, donor_data, out=sys.stdout):
+    """Record new donation and prints a report.
+
+    Args:
+        name (str)
+        num (int)
+        donor_data (dict)
+        out (file, optional)
+    """
+    donor_data[name].append(num)
+    message = u'Thanks {} for your generous donation of {} dollars.\n'
+    out.write(message.format(name, num))
+
+
+def align_cell(value, padding):
+    """Align a single value using padding.
+
+    Args:
+        value (str or int)
+        padding (int)
+    """
+    whitespace = u' ' * (padding - len(str(value)) + 2)
+    return u'{}{}'.format(value, whitespace)
```

Comment:
look into the string formatting specification for an easier way of accomplishing this.